crates/sovereign-sdk/full-node/sov-stf-runner/src/prover_service/parallel/mod.rs

```diff
@@ -167,12 +173,16 @@ where
                 ProverStatus::Proved(proof) => {
                     let da_data = DaData::ZKProof(proof.clone());
 
-                    let tx_id = da_service
-                        .send_transaction(
-                            da_data.try_to_vec().expect("Should serialize").as_slice(),
-                        )
-                        .await
-                        .map_err(|e| anyhow::anyhow!(e))?;
+                    let tx_id = backoff::future::retry(exponential_backoff, || async {
+                        da_service
+                            .send_transaction(
+                                da_data.try_to_vec().expect("Should serialize").as_slice(),
```

Comment:
@kpp given that we retry sending txs at bitcoin-da level, does it make sense to retry here?

Reply:
We don't seem to be retrying sending transactions.

The `send_transaction` functions sends to the bitcoin-da receiver loop to `send_transaction_with_fee` which would happen only once and would not be retried.

1.
https://github.com/chainwayxyz/citrea/blob/2e7fa5afc35cdccf9938fe51af6b8261c78654c7/crates/bitcoin-da/src/service.rs#L502-L513

2.
https://github.com/chainwayxyz/citrea/blob/2e7fa5afc35cdccf9938fe51af6b8261c78654c7/crates/bitcoin-da/src/service.rs#L502-L513

3
https://github.com/chainwayxyz/citrea/blob/2e7fa5afc35cdccf9938fe51af6b8261c78654c7/crates/bitcoin-da/src/service.rs#L106-L138

Reply:
see in 3, if there is an error from any rpc calls, the loop continues to try

Reply:
oh i see your point, we have
```
while receive {
   loop {}
}

Reply:
It doesn't make sense to retry outside since the outside async task would retry sending through the channel in this case.

Reply:
Removed